### PR TITLE
fix: Images do not load after restore of history backup (WEBAPP-5004)

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: app/script/backup/BackupRepository.js [hunk 1]
```diff
@@ -78,12 +78,6 @@ z.backup.BackupRepository = class BackupRepository {
           throw new z.backup.CancelError();
         }
         progressCallback(rows.length);
-
-        const isEventTable = tableName === z.storage.StorageSchemata.OBJECT_STORE.EVENTS;
```
Comment (atomrc) [inline app/script/backup/BackupRepository.js]: Wow, this is not normal at all. This is something we added this morning, it's not supposed to be removed

Comment (atomrc) [reply app/script/backup/BackupRepository.js]: There must have been a weird git operation I think

Comment (AndyLnd) [reply app/script/backup/BackupRepository.js]: weird ... i will add it back in

**Comment by AndyLnd** [comment]
When the database content gets JSON.stringify-ed it cannot stringify Uint8Arrays and keep the types. So we have to cast them as such when importing them. Otherwise they won't work anymore.

**Comment by atomrc** [comment]
Alright, thanks for the explanation @AndyLnd :+1:

File: app/script/backup/BackupRepository.js [hunk 2]
```diff
@@ -138,6 +138,18 @@ z.backup.BackupRepository = class BackupRepository {
       );
     });
 
+    const mapEntityDataTypes = entity => {
+      const mappedFields = ['otr_key', 'sha256'];
+      if (entity.data) {
+        mappedFields.forEach(field => {
+          if (entity.data[field]) {
+            entity.data[field] = Uint8Array.from(Object.values(entity.data[field]));
```
Comment (gregor) [inline app/script/backup/BackupRepository.js]: Please check that Object.values is supported by our browser range. I recall that we had an issue with this and outdated wrappers in the wild at some point in the past.

File: app/script/backup/BackupRepository.js [hunk 3]
```diff
@@ -138,6 +138,18 @@ z.backup.BackupRepository = class BackupRepository {
       );
     });
 
+    const mapEntityDataTypes = entity => {
+      const mappedFields = ['otr_key', 'sha256'];
```
Comment (gregor) [inline app/script/backup/BackupRepository.js]: I suppose we can move these out of here into a static config const.

File: app/script/backup/BackupRepository.js (hunk 2 above)
Comment (AndyLnd) [reply app/script/backup/BackupRepository.js]: Good call. I will check.

**Pull request merged**